toolkit/tools/internal/miscutils/miscutils.go

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package miscutils
+
+import "crypto/rand"
+
+// RandomString generates a random string of the length specified
+// using the provided legalCharacters.  crypto.rand is more secure
+// than math.rand and does not need to be seeded.
+func RandomString(length int, legalCharacters string) (output string, err error) {
+	b := make([]byte, length)
+	_, err = rand.Read(b)
+	if err != nil {
+		return
+	}
+
+	count := byte(len(legalCharacters))
```

Comment:
>byte(len(legalCharacters)) [](start = 10, length = 26)

What if we pass more legal characters than `byte` can handle? Also, how do we handle characters, which take up more than one byte?

Reply:
Not sure the rational behind using `byte` here actually, an int should suffice... We are just using it as an index into a string.

Reply:
I also don't think `byte` has any justification. And since for this function we care about runes, not bytes, then `len()` is not the right way to go in a general case.

---
In reply to: [549888966](https://github.com/microsoft/CBL-Mariner/pull/479#discussion_r549888966) [](ancestors = 549888966)

Reply:
Wrote a new random string function along with some test cases for it.

Reply:
https://github.com/microsoft/CBL-Mariner/pull/479/commits/c1ed726db4928c43b8b543793e1724a955d1e70e